keps/487-kubectl-plugin/README.md

```diff
@@ -0,0 +1,573 @@
+`kubectl` and don't want to learn an entirely new means to interact with workloads.
+The proposed plugin makes the transition much easier for me.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+Maintaining the plugin in-sync with the latest version of Kueue will add additional
+maintainer responsibilities.
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Summary
+
+Kueuectl is a command-line tool for interacting with Kueue, and can serve as a standalone tool or be renamed/installed to act as a kubectl plugin. Using `kueuectl` a user can manage workloads. An alternative (shorter and easier to type name) might also be `kuctl`, which I'll use for the remainder of this document. Thus, the main client takes the following format:
+
+```bash
+kuctl [subcommand] [flags]
+```
+
+As a kubectl plugin, we would install it named as `kubectl-kueue` and then the interaction would be:
+
+```bash
+kubectl kueue [subcommand] [flags]
+```
+
+We can start with basic query of workload metadata and status, and move toward a tool that can further create / delete or otherwise interact with workloads. This design document will proceed with proposed interactions and example tables, which would be printed in the terminal.
+
+
+### Version
+
+```bash
+kuctl version
+```
+
+### Listing Workloads
+
+The most likely thing a user wants to do is see "all" workloads. We will need to decide if "all" means all namespaces, or (akin to `kubectl` just those in default.  I think likely this set of attributes needs a queue too. I'm undecided if we should enforce the user to provide the queue as the first argument, e.g,
+
+```bash
+kuctl workloads <queue-name>
+```
+
+| Name     | API Group         | Kind        | Namespace | Command | Pods | Time   |     State |
+|----------|-------------------|-------------|-----------|---------|------|--------|-----------|
+| taco-345 | flux-framework.org| MiniCluster | default   | python  | 3    | 2.322s |   Running | 
+| taco-123 | flux-framework.org| MiniCluster | default   | python  | 2    |        |   Pending | 
```

Comment:
I guess we could show the owner Kind from the owner reference.
But while that is ok for MiniCluster, is that generally ok? Should we show both?

Reply:
I think Kind carries a lot of meaning - MiniCluster / MPIJob are meaningful to me. API Group is less so, at least for this use case was the first time I looked up `kubeflow.org` and I only knew the flux one because I develop the flux operator.